src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -2490,21 +2490,14 @@ private void VisitForStatement(ParseNode node)
             // jump to condition
             Opcode jump = AddOpcode(new OpcodeBranchJump());
             jump.DestinationLabel = conditionLabel;
-            // end of loop, cleanup
-            Opcode endLoop = AddOpcode(new OpcodePush(iteratorIdentifier));
-            AddOpcode(new OpcodePush("reset"));
-            AddOpcode(new OpcodeGetMember());
-            AddOpcode(new OpcodePop()); // removes the "true" returned by the previous getmember
-            // unset of iterator and iteration variable
-            AddOpcode(new OpcodePush(iteratorIdentifier));
-            AddOpcode(new OpcodeUnset());
-            VisitVariableNode(node.Nodes[1]);
-            AddOpcode(new OpcodeUnset());
```

Comment:
This code made sense when variables were all global..  But now, there's no point in doing cleanup on it when it's about to go out of scope on the very next instruction, and get orphaned away.